test/unit/resources/virtualization_test.rb

```diff
@@ -12,17 +12,7 @@ def mock_proc(mocked_files)
     end
   end
 
-  it "fails the resource if OS is not Linux" do
-    mocked_files = []
-    mock_loader = MockLoader.new(:windows)
-
-    mock_loader.backend.stub :file, mock_proc(mocked_files) do
-      mock_resource = mock_loader.load_resource("virtualization")
-      _(mock_resource.resource_failed?).must_equal true
-    end
-  end
-
-  it "returns nil for all properties if no virutalization platform is found" do
+  it "returns nil for all properties if no virtualization platform is found" do
```

Comment:
The full set of tests here would be testing Linux and Windows; for each of the virtualisation systems (at least each that has a distinct execution flow), plus a test for detecting physical on both Linux and Windows.

That is quite a lot of work and involves creating new data for the mock loader. 
I think it is acceptable to merge this as-is, but I will update the resource improvement epic https://github.com/inspec/inspec/issues/4782 and comment that this resource needs testing filled out.

Reply:
Actually I tried figuring out how to mock file contents and such and had to give up after a while.

Maybe some documentation on how to properly write unit tests and the different patterns would be an idea, showing examples for the use cases instead of needing to dig through the literally thousands of tests (respect!). I suspect that this is mostly straighforward for those who wrote those tests.

Reply:
I agree, we need to improve this.
Documenting the patterns and hierarchy of the current tests is a great idea. Or refactor them so that the folder hierarchy and file and test names make the tests self-describing.